floris/tools/floris_interface.py

```diff
@@ -108,175 +111,134 @@ def copy(self):
+            wind_shear=wind_shear,
+            wind_veer=wind_veer,
+            reference_wind_height=reference_wind_height,
+            turbulence_intensities=turbulence_intensities,
+            air_density=air_density,
+            layout_x=layout_x,
+            layout_y=layout_y,
+            turbine_type=turbine_type,
+            turbine_library_path=turbine_library_path,
+            solver_settings=solver_settings,
+            heterogenous_inflow_config=heterogenous_inflow_config,
+            wind_data=wind_data,
+        )
+        if not (save_yaw_angles == 0).all():
+            self.floris.farm.yaw_angles = save_yaw_angles
+        if not (
+            (save_power_setpoints == POWER_SETPOINT_DEFAULT)
+            | (save_power_setpoints == POWER_SETPOINT_DISABLED)
+        ).all():
+            self.floris.farm.power_setpoints = save_power_setpoints
+
+        # Set the operation
+        self._set_operation(
+            yaw_angles=yaw_angles,
+            power_setpoints=power_setpoints,
+            disable_turbines=disable_turbines,
+        )
 
-        if yaw_angles is None:
-            yaw_angles = np.zeros(
-                (
-                    self.floris.flow_field.n_findex,
-                    self.floris.farm.n_turbines,
-                )
-            )
-        self.floris.farm.yaw_angles = yaw_angles
-
-        if power_setpoints is None:
-            power_setpoints = POWER_SETPOINT_DEFAULT * np.ones(
-                (
-                    self.floris.flow_field.n_findex,
-                    self.floris.farm.n_turbines,
-                )
-            )
-        else:
-            power_setpoints = np.array(power_setpoints)
-
-        # Convert any None values to the default power setpoint
-        power_setpoints[
-            power_setpoints == np.full(power_setpoints.shape, None)
-        ] = POWER_SETPOINT_DEFAULT
-        power_setpoints = floris_array_converter(power_setpoints)
-
-        # Check for turbines to disable
-        if disable_turbines is not None:
-
-            # Force to numpy array
-            disable_turbines = np.array(disable_turbines)
-
-            # Must have first dimension = n_findex
-            if disable_turbines.shape[0] != self.floris.flow_field.n_findex:
-                raise ValueError(
-                    f"disable_turbines has a size of {disable_turbines.shape[0]} "
-                    f"in the 0th dimension, must be equal to "
-                    f"n_findex={self.floris.flow_field.n_findex}"
-                )
-
-            # Must have first dimension = n_turbines
-            if disable_turbines.shape[1] != self.floris.farm.n_turbines:
-                raise ValueError(
-                    f"disable_turbines has a size of {disable_turbines.shape[1]} "
-                    f"in the 1th dimension, must be equal to "
-                    f"n_turbines={self.floris.farm.n_turbines}"
-                )
+    def reset_operation(self):
```

Comment:
If this function only calls another function do we still need it?  Is it a public private split?

Reply:
Good point. I'm not sure how to handle it. `_reinitialize` does set the operation set-points back to their defaults, but it's also very close to the v3 `reinitialize` so I think it's a bad idea to keep that as a public function if we want to move to a different workflow.